### PR TITLE
Use CopyImageBitmapToTexture for webgpu backend

### PR DESCRIPTION
**Pull request by shaoboyan**
ImageBitmap is also a possible input from HTML. Adding ImageBitmap as
fromPixels valid input type.

CopyImageBitmapToTexture is usual API for uploading in WebGPU.
Integrate this API into fromPixels in tfjs webgpu backend.

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4374)

### CODE REVIEW

**Comment by shaoboyan** [comment]
@qjia7  PTAL.

File: tfjs-core/src/ops/from_pixels_test.ts
Comment (qjia7) [inline tfjs-core/src/ops/from_pixels_test.ts:301]: Can you also check if data is expected to make sure `fromPixels` is correctly executed?

**Review by qjia7 — COMMENTED** [review]
LGTM with a nit.
@pyu10055 @annxingyuan @kainino0x take a look. Thanks.

**Comment by qjia7** [comment]
I prefer you change your title to `make ImageBitmap as an acceptable type for fromPixels`. You even can split `Use CopyImageBitmapToTexture for webgpu backend` part from this patch for easier review.

File: tfjs-core/src/ops/from_pixels_test.ts
Comment (kainino0x) [inline tfjs-core/src/ops/from_pixels_test.ts:302]: nit: close imageBitmap immediately after fromPixels, to make sure it's okay to do so (unless it's not)

File: tfjs-backend-webgpu/src/kernels/FromPixelsImageBitmap.ts
Comment (kainino0x) [inline tfjs-backend-webgpu/src/kernels/FromPixelsImageBitmap.ts:42]: Is caching of the pipeline object done inside compileProgram? Otherwise this would cause a lot of recompilations if the outputShape isn't steady.

**Review by kainino0x — APPROVED** [review]

File: tfjs-backend-webgpu/src/kernels/FromPixelsImageBitmap.ts
Comment (shaoboyan) [reply tfjs-backend-webgpu/src/kernels/FromPixelsImageBitmap.ts]: Yes. You're right and thanks for catching this. Now add an API to update the outShape for FromPixelsProgram. Please take a look

File: tfjs-core/src/ops/from_pixels_test.ts
Comment (shaoboyan) [reply tfjs-core/src/ops/from_pixels_test.ts:301]: Done.

File: tfjs-core/src/ops/from_pixels_test.ts
Comment (kainino0x) [inline tfjs-core/src/ops/from_pixels_test.ts:305]: nit: all of these `var`s can be `const`s I think. (Any that can't should be `let` instead.)

File: tfjs-backend-webgpu/src/kernels/FromPixelsImageBitmap.ts
Comment (kainino0x) [inline tfjs-backend-webgpu/src/kernels/FromPixelsImageBitmap.ts:47]: nit: I think this check should be inside updateOutputShape

**Review by kainino0x — APPROVED** [review]

File: tfjs-backend-webgpu/src/kernels/FromPixels_utils/from_pixels_webgpu.ts
Comment (kainino0x) [inline tfjs-backend-webgpu/src/kernels/FromPixels_utils/from_pixels_webgpu.ts:79]: Code looks duplicated with constructor. Call this from the constructor instead of having the code twice?

File: tfjs-backend-webgpu/src/kernels/FromPixelsImageBitmap.ts
Comment (kainino0x) [reply tfjs-backend-webgpu/src/kernels/FromPixelsImageBitmap.ts:47]: Actually, I think this check is also wrong. You can't do `==`/`!=` or `===`/`!==` between two arrays, it compares them by reference, not by value.

**Review by kainino0x — CHANGES_REQUESTED** [review]

File: tfjs-backend-webgpu/src/kernels/FromPixelsImageBitmap.ts
Comment (shaoboyan) [reply tfjs-backend-webgpu/src/kernels/FromPixelsImageBitmap.ts]: @kainino0x  thanks for pointing this out. Please take another look when you're available.

File: tfjs-backend-webgpu/src/kernels/FromPixels_utils/from_pixels_webgpu.ts
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/FromPixels_utils/from_pixels_webgpu.ts:44]: Nit: you can use `util.arraysEqual(this.outputShape, outputShape)`

File: tfjs-core/src/ops/from_pixels_test.ts
Comment (qjia7) [inline tfjs-core/src/ops/from_pixels_test.ts:292]: Use `if (i % 4 === 3)` to fix the bot error.

**Review by kainino0x — APPROVED** [review]
LGTM after Jiajia's comments

**Review by kainino0x — APPROVED** [review]
LGTM, but build is failing. Not sure why; may or may not be unrelated to this change. I think it timed out while testing in Chrome?

> Step #47 - "test-tfjs":   Disconnected, because no message in 30000 ms.

**Comment by shaoboyan** [comment]
@kainino0x  thanks for helping. I double checked the tests on my local machine and it can finish. So I suspect the bots failure is not related to my patch.
@pyu10055 @annxingyuan  could you help to fix the bots ? Thanks!

**Comment by qjia7** [comment]
@tafsiri is actively reviewing #4375 which may need this patch's support.

**Comment by shaoboyan** [comment]
@tafsiri @kainino0x @pyu10055  This CL has been rebased  and please help to merge this pr, thx!

**Comment by shaoboyan** [comment]
@tafsiri @kainino0x @pyu10055  ping softly for merge this pr, thx!

**Review by kainino0x — APPROVED** [review]
Sorry, Google was out yesterday. Still LGTM :)

**Comment by kainino0x** [comment]
I have permission to merge but I usually let TF.js maintainers do so. We can chat about the procedure at our meeting.

**Comment by shaoboyan** [comment]
@kainino0x  Thank you very much!

**Comment by shaoboyan** [comment]
The bots failed due to timeout. I think I need to re-run it. (And I don't know any possible method to re-try it by myself except fo "update branch" button or do push -f).

**Comment by shaoboyan** [comment]
@kainino0x  Thanks for merging this!

**Pull request merged**
